06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * An inverted-index class
+ * @class
+ */
+class InvertedIndex {
+  /**
+  * class constructor
+  * @constructor
+  */
+  constructor() {
+    this.index = {};
+    this.object = {};
+    this.allFiles = {};
+    this.allLength = {};
+    this.allFilesTitle = [];
+  }
+  /**
+   * Gets Indexes
+   * @param {jsonObj} object receives json object
+   * @param {string} fileName of the books
+   * @return {object} returns an object of Indexes
+   */
+  createIndex(object, fileName) {
```

Comment:
I think this method is doing too much. In here you're sanitising inputs and then building the index from the files. Let's break this down.